src/video-custom-controls.js

```diff
@@ -0,0 +1,687 @@
+import {listen} from './event-helper';
+import {Services} from './services';
+import {VideoEvents} from './video-interface';
+import {secsToHHMMSS} from './utils/datetime';
+import * as st from './style';
+import * as tr from './transition';
+
+
+/**
+ * CustomControls is a class that given a video manager entry
+ * ({@link ./service/video-manager-impl.VideoEntry}), adds an overlay of
+ * customizable controls to the video element and manages their behavior.
+ */
+export class CustomControls {
+
+  /**
+   * Initializes variables and creates the custom controls
+   * @param {!./service/ampdoc-impl.AmpDoc} ampdoc
+   * @param {!./service/video-manager-impl.VideoEntry} entry
+   * @param {{
+   *    darkSkin:(boolean|undefined),
+   *    mainCtrls:(Array<string>|undefined),
+   *    miniCtrls:(Array<string>|undefined),
+   *    floating:(string|undefined),
+   * }=} opt_options
+   *      - darkSkin: whether to use dark or light theme
+   *      - mainCtrls: list of controls to add to the main bar
+   *      - miniCtrls: list of controls to add to the minimized overlay
+   *      - floating: single control button to use as the main action
+   */
+  constructor(ampdoc, entry, opt_options) {
+
+    /** @private {!./service/ampdoc-impl.AmpDoc}  */
+    this.ampdoc_ = ampdoc;
+
+    /** @private {!./service/video-manager-impl.VideoEntry} */
+    this.entry_ = entry;
+
+    /** @private @const {!./service/vsync-impl.Vsync} */
+    this.vsync_ = Services.vsyncFor(this.ampdoc_.win);
+
+    this.options_ = opt_options || {
+      darkSkin: false,
+      mainCtrls: ['time', 'spacer', 'volume', 'fullscreen'],
+      miniCtrls: ['play', 'volume', 'fullscreen'],
+      floating: 'play',
+    };
+
+    /** @private {?Element} */
+    this.ctrlContainer_ = null;
+
+    /** @private {?Element} */
+    this.ctrlBarContainer_ = null;
+
+    /** @private {?Element} */
+    this.ctrlBarWrapper_ = null;
+
+    /** @private {?Element} */
+    this.floatingContainer_ = null;
+
+    /** @private {?Element} */
+    this.miniCtrlsContainer_ = null;
+
+    /** @private {?Element} */
+    this.miniCtrlsWrapper_ = null;
+
+    /** @private {?Element} */
+    this.ctrlBg_ = null;
+
+    /** @private {?number} */
+    this.controlsTimer_ = null;
+
+    /** @private {boolean} */
+    this.controlsShown_ = true;
+
+    /** @private {boolean} */
+    this.controlsShowing_ = false;
+
+    /** @private {boolean} */
+    this.controlsDisabled_ = false;
+
+    /** @private {boolean} */
+    this.minimal_ = false;
+
+    this.createCustomControls_(this.options_);
+  }
+
+  /**
+   * Returns the element that wraps all the controls
+   * @return {!Element}
+   */
+  getElement() {
+    return dev().assertElement(this.ctrlContainer_);
+  }
+
+  /**
+   * Fullscreen Button
+   * @return {!Element}
+   * @private
+   */
+  createFullscreenBtn_() {
+    const doc = this.ampdoc_.win.document;
+    const fsBtnWrap = doc.createElement('amp-custom-controls-icon-wrapper');
```

Comment:
I'd make these `divs` and just use css classes.
also let's add `media` to the name.

fsBtn.class = `amp-media-custom-controls-icon-wrapper amp-media-custom-controls-fullscreen`